File: src/main/java/org/taktik/icure/logic/CalendarItemTypeLogic.java
package org.taktik.icure.logic;

import org.taktik.icure.entities.CalendarItemType;
import org.taktik.icure.exceptions.DeletionException;

import java.util.List;

public interface CalendarItemTypeLogic extends EntityPersister<CalendarItemType, String> {
    CalendarItemType createCalendarItemType(CalendarItemType calendarItemType);

    List<String> deleteCalendarItemTypes(List<String> ids) throws DeletionException;

    CalendarItemType getCalendarItemType(String calendarItemTypeId);

    CalendarItemType modifyCalendarTypeItem(CalendarItemType calendarItemType);

    List<CalendarItemType> getAllEntitiesIncludeDelete();
}

File: src/main/kotlin/org/taktik/icure/asynclogic/CalendarItemTypeLogic.kt
/*
 *  iCure Data Stack. Copyright (c) 2020 Taktik SA
 *
 *     This program is free software: you can redistribute it and/or modify
 *     it under the terms of the GNU Affero General Public License as
 *     published by the Free Software Foundation, either version 3 of the
 *     License, or (at your option) any later version.
 *
 *     This program is distributed in the hope that it will be useful, but
 *     WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 *     Affero General Public License for more details.
 *
 *     You should have received a copy of the GNU Affero General Public
 *     License along with this program.  If not, see
 *     <https://www.gnu.org/licenses/>.
 */

package org.taktik.icure.asynclogic

import kotlinx.coroutines.flow.Flow
import org.taktik.couchdb.DocIdentifier
import org.taktik.icure.asyncdao.CalendarItemTypeDAO
import org.taktik.icure.entities.CalendarItemType

interface CalendarItemTypeLogic : EntityPersister<CalendarItemType, String> {
	suspend fun createCalendarItemType(calendarItemType: CalendarItemType): CalendarItemType?
	fun deleteCalendarItemTypes(ids: List<String>): Flow<DocIdentifier>

	suspend fun getCalendarItemType(calendarItemTypeId: String): CalendarItemType?

	suspend fun modifyCalendarTypeItem(calendarItemType: CalendarItemType): CalendarItemType?
	fun getAllEntitiesIncludeDelete(): Flow<CalendarItemType>
	fun getGenericDAO(): CalendarItemTypeDAO
	fun getCalendarItemTypes(calendarItemTypeIds: Collection<String>): Flow<CalendarItemType>
}
